qa/vagrant/src/test/resources/packaging/scripts/plugin_test_cases.bash

```diff
@@ -0,0 +1,193 @@
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+##################################
+# Common test cases for both tar and rpm/deb based plugin tests
+##################################
+# This file is symlinked to both 25_tar_plugins.bats and 50_plugins.bats so its
+# executed twice - once to test plugins using the tar distribution and once to
+# test files using the rpm distribution or the deb distribution, whichever the
+# system uses.
+
+# Load test utilities
+load packaging_test_utils
+load plugins
+
+setup() {
+    # The rules on when we should clean an reinstall are complex - all the
+    # jvm-example tests need cleaning because they are rough on the filesystem.
+    # The first and any tests that find themselves without an ESHOME need to
+    # clean as well.... this is going to mostly only happen on the first
+    # non-jvm-example-plugin-test _and_ any first test if you comment out the
+    # other tests. Commenting out lots of test cases seems like a reasonably
+    # common workflow.
+    if [ $BATS_TEST_NUMBER == 1 ] ||
+            [[ $BATS_TEST_NAME =~ install_jvm.*example ]] ||
+            [ ! -d "$ESHOME" ]; then
+        echo "cleaning" >> /tmp/ss
+        clean_before_test
+        install
+    fi
+}
+
+if [[ "$BATS_TEST_FILENAME" =~ 25_tar_plugins.bats$ ]]; then
+    load tar
+    GROUP='TAR PLUGINS'
+    install() {
+        install_archive
+        verify_archive_installation
+    }
+    export ESHOME=/tmp/elasticsearch
+    export_elasticsearch_paths
+else
+    if is_rpm; then
+        GROUP='RPM PLUGINS'
+    elif is_dpkg; then
+        GROUP='DEB PLUGINS'
+    fi
+    export ESHOME="/usr/share/elasticsearch"
+    export ESPLUGINS="$ESHOME/plugins"
+    export ESCONFIG="/etc/elasticsearch"
+    install() {
+        install_package
+        verify_package_installation
+    }
+fi
+
+@test "[$GROUP] install jvm-example plugin" {
+    install_jvm_example
+    remove_jvm_example
+}
+
+@test "[$GROUP] install jvm-example plugin with a custom path.plugins" {
+    # Clean up after the last time this test was run
+    rm -rf /tmp/plugins.*
+
+    local oldPlugins="$ESPLUGINS"
+    export ESPLUGINS=$(mktemp -d -t 'plugins.XXXX')
+
+    # Modify the path.plugins setting in configuration file
+    echo "path.plugins: $ESPLUGINS" >> "$ESCONFIG/elasticsearch.yml"
+    chown -R elasticsearch:elasticsearch "$ESPLUGINS"
+
+    install_jvm_example
+    remove_jvm_example
+}
+
+@test "[$GROUP] install jvm-example plugin with a custom CONFIG_DIR" {
+    # Clean up after the last time we ran this test
+    rm -rf /tmp/config.*
+
+    move_config
+
+    CONF_DIR="$ESCONFIG" install_jvm_example
+    CONF_DIR="$ESCONFIG" remove_jvm_example
+}
+
+@test "[$GROUP] install jvm-example plugin with a custom ES_JAVA_OPTS" {
```

Comment:
There's good chance that this test becomes obsolete soon with https://github.com/elastic/elasticsearch/pull/12801... I'm +1 for removing it now